docs/pages/ds-video-port_High-Level_TestSpec.md

```diff
@@ -0,0 +1,182 @@
+
+### Emulator Requirements
+
+Boot configuration: Check video ports, formats, frame rates, and device-supported resolutions, along with the number of ports supported by each device.
+
+Supported Video Port [dsVideoPortType_t link](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L132)
+
+Supported Video resolutions [dsTVResolution_t link](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L489)
+
+Supported Frame rates [dsVideoFrameRate_t link](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L508)
+
+Supported Video formats [dsHDRStandard_t link](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L625)
+
+Supported Video background Color [dsVideoBackgroundColor_t link](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L540)
+
+Supported Display Color Depth [dsDisplayColorDepth_t link](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L758)
+
+Supported Display Color Space [dsDisplayColorSpace_t link](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L733)
+
+Supported Display MatrixCoefficient [dsDisplayMatrixCoefficients_t link](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L780)
+
+Supported Display QuantizationRange [dsDisplayQuantizationRange_t link](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L745)
+
+### Check the video port status
+
+|Test Functionality|Description|L2|L3|Source|Sink|
+|------------------|-----------|--|--|------|----|
+|Check the video port status|Get the video port handle,check the Video Port enable/disable|`Y`|`Y`|`Y`|`Y`|
+||Verify display connected/disconnected status without video port connected|`Y`|`NA`|`Y`|`Y`|
+||Verify display connected/disconnected status by connecting/disconnecting video port|`NA`|`Y`|`Y`|`NA`|
+||Verify the surround mode capabilities with/without playback|`Y`|`Y`|`Y`|`Y`|
+||Verify the surround mode capabilities of connected display with/without playback|`Y`|`Y`|`Y`|`NA`|
+
+#### Test Startup Requirement-Check the video port status
+
+Playback the pre-define streams
+
+#### Emulator Requirements-Check the video port status
+
+[Emulator Requirements](#emulator-requirements)
+
+#### Control Plane Requirements-Check the video port status
+
+Unplug the Video port,Verify the surround mode
+
+### Check Video Content Format and HDR Capability
+
+|Test Functionality|Description|L2|L3|Source|Sink|
+|------------------|-----------|--|--|------|----|
+|Check Video Format Content and HDR Capability|Notify an event when the list of video Format changes|`NA`|`Y`|`Y`|`Y`|
+||Get the HDR capabilities and status|Y|`NA`|`Y`|`Y`|
+||Set Force Disable 4KSupport with/without playback|`Y`|`Y`|`Y`|`Y`|
+||Disable 4KSupport with external analyzer|`NA`|`Y`|`Y`|`NA`|
```

Comment:
Same here

Reply:
if the interface header has API we need to write a test case for that API. if the device/ platform not supported then as you said it will return NOT supported or dsERR_OPERATION_NOT_SUPPORTED.

Reply:
Fine, API returns dsERR_OPERATION_NOT_SUPPORTED.